Test/Utils/ContrailAPI/DNSServerRepo_Impl.ps1

```diff
@@ -0,0 +1,98 @@
+class DNSServerRepo {
+    [String] $ServerResourceName = "virtual-DNS"
+    [String] $RecordResourceName = "virtual-DNS-record"
+
+    [ContrailNetworkManager] $API
+
+    DNSServerRepo ([ContrailNetworkManager] $API) {
+        $this.API = $API
+    }
+
+    [String] AddContrailDNSServer([DNSServer] $DNSServer) {
+        $VirtualDNS = @{
+            domain_name                 = $DNSServer.DomainName
+            dynamic_records_from_client = $DNSServer.DynamicRecordsFromClient
+            record_order                = $DNSServer.RecordOrder
+            default_ttl_seconds         = $DNSServer.DefaultTTLInSeconds
+            floating_ip_record          = $DNSServer.FloatingIpRecord
+            external_visible            = $DNSServer.ExternalVisible
+            reverse_resolution          = $DNSServer.ReverseResolution
+        }
+
+        $Request = @{
+            "virtual-DNS" = @{
+                parent_type      = "domain"
+                fq_name          = $DNSServer.GetFQName()
+                virtual_DNS_data = $VirtualDNS
+            }
+        }
+
+        $Response = $this.API.SendRequest("Post", $this.ServerResourceName, $null, $Request)
+        $DNSServer.uuid = $Response.'virtual-DNS'.'uuid'
+        return $DNSServer.uuid
+    }
+
+    RemoveContrailDNSServer([DNSServer] $DNSServer, [Boolean] $Force) {
```

Comment:
Maybe we should introduce two separate methods: `RemoveContrailDNSServer` and `RemoveContrailDNSServerWithDependencies`?

`RemoveContrailDNSServer` would be current `RemoveContrailDNSServer` with `$Force` set to `$False`, and `RemoveContrailDNSServerWithDependencies` would have `$Force` set to `$False`.

Also I think we can remove `Contrail` from method names.